### ISSUE TITLE
Enhance "Get started" and other vignettes

### ISSUE THREAD

**Issue by krlmlr** [ropenscilabs/tic#78]
@pat-s: I like the way the new vignettes are organized, they are good read.
The relevant information is presented concisely, with good cross-references.
Especially the automatic linking from code chunks to function documentation done by _pkgdown_ makes the vignettes easy to navigate.

The vignettes could use some graphics, screenshots, and code blocks/listings.
They could be a bit more self-contained, too -- e.g. show an example instead of a bare link pointing to additional information.
The examples are important to give the user an early understanding if following the link is worthwhile.

# README

- [x] Link to CI intro (mahugh.com), brief before-after comparison (with image from ) and link to "advantages" website
- [x] "Example projects" is anchored link, just mention when introducing the "Get started" vignette?
- [x] Footer on website?

# Get started

- [x] Prerequisites: Second paragraph lengthy and diluted; instead briefly summarize tic's advantages in a comprehensive bullet list (one item per benefit, like in the intro of the "Advantages" vignette), add a before-after like in README, and link to `tic-advantages.html`
- [x] Initialization: Again, can be a bit longer, showing the call and the expected output in a full-width chunk. Mention that browser pages will open and app authorization is required, and a PAT needs to be created, add screenshots. Explain why this is necessary, especially that PAT creation can't be automated currently.
- Setup
    - [x] Consider renaming section
    - [x] Show and explain example `tic.R` file (copy template to the package for now, it might stay there for good, or we take the template from _usethis_ later)
    - [x] Don't show example `.travis.yml` and `appveyor.yml` files, but point to the "Build lifecycle" article
- [x] Examples
    - [x] In the bullet list, add a brief explanation, e.g. for blogdown: what is blogdown, how is deployment achieved, is there anything special about the example repo, maybe show excerpts from `tic.R`?
    - [x] The link names should match the repo names
    - [x] figshare publishing doesn't work yet, publishing to Zenodo is work in progress (need to ask Karthik)
    - [x] CTA: share your own nonstandard _tic_-based solutions?
- [x] What's not covered yet: move to "Advanced"

# Advantages

- [x] Title is repeated on rendered website?
- More advantages
    - [x] Easier troubleshooting for local builds (with link); it's mentioned in "CI-Agnostic" but deserves its own section with a working example
    - [x] Less dependent on code we don't control (e.g. `r.rb`)
- CI-agnostic:
    - [x] `tic.R` is shorter and easier to parse than `.travis.yml`
    - [x] AppVeyor is running on Linux since recently
- Easier deployment:
    - [x] Formatting: Indent code blocks and subsequent paragraphs with four spaces to make them indented in the enumeration
    - [x] Shorten, keep first paragraph and link to new "How deployment works" vignette
- R CMD check:
    - [x] Section title unclear
    - [x] _rcmdcheck_  calls into `R CMD build` and `R CMD check` (it's not reimplementing it), parses output and returns errors/warnings to the caller as an R object
    - [x] Show excerpts of colored output (use `fansi::sgr_to_html()` to create HTML)
- Caching:
    - [x] Link to "Build lifecycle"
    - [x] Build deps and check deps are stored in two separate libraries, both are cached. Not sure how to frame it, let's discuss.

# tic, travis, and usethis

Pending decision on where setup code is hosted.

Overall, I think a detailed walkthrough (expanding on what will be shown in "Get started") can add some value here.

- [x] Explain all the steps of `use_ci()` (whether its in tic or usethis) to see how the packages interact

In "Getting Started we only talk on a meta level about the steps of `use_ci()` and refer to the tic+travis+usethis vignette explaining the steps in detail.
    
# Build lifecycle

- Stages:
    - [x] Article title repeated?
    - [x] Need diagram
    - [x] Mention that `.travis.yml` is not meant to be edited (include the "DO NOT EDIT" comments in the code)
    - [x] ~~How is a stage accessed from R code? Show an example `tic.R` file and the output of e.g. `get_stage("script")`~~ How is a stage accessed from R code and how can a single stage be run?
- Steps:
    - [x] Mention macros (#61, to be implemented)
    - [x] Add steps not mentioned there yet
    - [x] Link steps to help pages, this should happen automatically if you add `()` after the step names
    - [x] Add more detail to the documentation of the steps

# New vignette: How deployment works

Derived from "Easier deployment" in the "Advantages" vignette, and "pkgdown deployment" and "committing single files" in "Advanced usage".

- [x] _tic_ creates public-private key pair
    - public key is added to GitHub repository under the "Deploy keys" section (screenshot), user receives e-mail
    - private key is encoded in base64 and stored in an environment variable, and installed in the `~/.ssh` directory during the CI run
- Prefer deployment via public-private key pair, because rights are granted to a single repo only and can be automated
    - We do create per-repo PATs anyway, but these have a very restrictive scope just to avoid rate limitation.
    - Need to revisit: perhaps creating a PAT with write permissions is easier if we can make sure that our scripts never leak the PAT?

# Advanced usage

- pkgdown and Committing single files:
    - [x] Move to "How deployment works"
- Running stages conditionally:
    - [x] Mention that `tic.R` is a file that is processed with `source()`
    - [x] Steps are added incrementally, existing steps created by macros aren't overwritten
    - [x] R version can be checked with `getRversion()`, show example where `warnings_are_errors` depends on the R version used (ignoring warnings for R 3.1)
- Installing packages:
    - [x] `s/add_code_step/add_step/`
    - [x] Mention that it shouldn't be used in interactive CI runs (`ci()$is_interactive()`)
    - [x] Packages should be installed from `DESCRIPTION`, install manually only in extraordinary circumstances
- Meta:
    - [x] Remove "Travis" from title
- Running tic locally:
    - [x] Show example output for `load_from_file()`
    - [x] Important: define steps so that they don't overwrite precious local state. Many steps are careful (double-check this), but custom code may be not
- Entering:
    - [x] Elaborate on opt-in, mention that it needs an e-mail which will be acted upon in less than one day usually.
    - [x] How to get the job ID? Show example in separate code chunk.
    - [x] Link to `tmate` key bindings documentation, mention similarity with `screen` and overarching Ctrl + A shortcut (and two or three most important ones, e.g. for scrolling back)
    - [x] Handy functions: Which ones?

# Dev info

- [x] "Most steps" -> "Many important steps". Always possible to add a code step, this text is only important for reocurring use if we want to add a new step to _tic_.
- [x] Point to PR that adds `step_install_cran()`
- [x] Show source code of the "Hello world" step
- [x] `check()` doesn't need to be overridden
- [x] `prepare()` is called by `prepare_all_stages()`
- [x] Show code of `RCMDcheck$prepare`
- [x] Start subsection before "If you are unfamiliar"

**Comment by pat-s** [comment]
> Footer on website?

I assume you mean to include the ropensci footer? Would be a bit redundant then if we keep the logo in the README.

**Comment by pat-s** [comment]
> Show excerpts of colored output (use fansi::sgr_to_html() to create HTML)

I tried to set it in the knitr header but to get it working we need to render code directly? I cannot really render `rcmdcheck()` in a tutorial page. How should we get the coloring right here?

```r
old.hooks = fansi::set_knit_hooks(knitr::knit_hooks)
options(crayon.enabled = TRUE)
```

**Comment by pat-s** [comment]
> Build deps and check deps are stored in two separate libraries, both are cached. Not sure how to frame it, let's discuss.

Here's what I wrote:

`tic` uses two distinct R libraries for caching: One for the dependencies that are used for checking the R package and another R package library for all other dependencies (e.g. for packages that are used during the "deploy" stage).
The reason for that is that sometimes dependencies for specific stages (e.g. "the deploy" stage) may cause errors when checking the R package because these contain breaking changes which the package is not able to deal with (e.g. if the Github version of a package is installed).
You do not need to worry about this in practice - both will be cached and handled in the background for you!

**Comment by pat-s** [comment]
> How is a stage accessed from R code? Show an example tic.R file and the output of e.g. get_stage("script")

I would not again show the rcmdcheck output of the script stage here as we already show this in the rcmdcheck section in the "advantage" vignette.

Instead I combined the question how a stage is accessed from R code with how to access a single stage.

**Comment by pat-s** [comment]
> Add more detail to the documentation of the steps

I would keep it concise and refer to the detailed help page of each step.

**Comment by pat-s** [comment]
> Important: define steps so that they don't overwrite precious local state. Many steps are careful (double-check this), but custom code may be not

I cannot think of a good example here. Which state might be so precious and at the same time "easily" overwritten by a user by accident?

**Comment by krlmlr** [comment]
- README: I restructured it (in your branch), works for me now.

- Footer looks good on website now

- Let's think about coloring later (#91)

**Comment by krlmlr** [comment]
Closed in #80.
